### PR TITLE
Dev generator

### PR DESCRIPTION
**Pull request by AndHager**
A further generator for a maze

### CODE REVIEW

File: install_dev_deps.sh [hunk 1]
```diff
@@ -1 +1 @@
-apt-get update && apt-get install -y libgl1-mesa-dev libgtk-3-dev libcairo2-dev libglib2.0-dev
+sudo apt-get update && sudo apt-get install -y libgl1-mesa-dev libgtk-3-dev libcairo2-dev libglib2.0-dev
```
Comment (electron2302) [inline install_dev_deps.sh:1]: I think the sudo should not be in the scipt, the user who runs that must have the rights to run the commands. The script shouldn’t take care of that

File: generator/generator.go [hunk 2]
```diff
@@ -1,7 +1,57 @@
...
 	GenerateLabyrinth(furthestPoint common.Location) (common.Labyrinth, []common.Pair)
 }
+
+const (
+	dCoordinate = 1
+	Start       = "START"
+	Discover    = "DISCOVER"
+	Backtrack   = "BACKTRACK"
+	Select      = "SELECT"
+	Add         = "ADD"
```
Comment (electron2302) [inline generator/generator.go:20]: Could you elaborate on the differences between Discoder, Selecte and Add

File: install_dev_deps.sh (hunk 1 above)
Comment (AndHager) [reply install_dev_deps.sh]: oke removed

File: generator/generator.go (hunk 2 above)
Comment (AndHager) [reply generator/generator.go:20]: DISCOVER means that the marked node is discovered and set as the current node for further (recursive) discovering of nodes.
SELECT means that a node is selected as a base do ADD all available neighbors
ADD means that this node was added by the last SELECTed node

Comment (AndHager) [reply generator/generator.go:20]: Would you suggest other (better) names?

Comment (electron2302) [reply generator/generator.go:20]: Thanks for the explanation

**Pull request merged**